packages/design-system/src/components/Form/Field/Input/Input.Checkbox.tsx

```diff
@@ -120,11 +120,16 @@ const Checkbox = React.forwardRef(
 	) => {
 		const { id: reakitId } = useId();
 		const checkboxId = id || `checkbox--${reakitId}`;
+		const state = (indeterminate && 'indeterminate') || defaultChecked || checked;
 		const checkbox = useCheckboxState({
-			state: (indeterminate && 'indeterminate') || defaultChecked || checked,
+			state,
 			readOnly,
 		});
 
+		useEffect(() => {
+			checkbox.setState(state);
+		}, [state]);
```

Comment:
I'm not sure this hook is valid, aren't we supposed to have the exhaustive dependancy list in the array? We'd be missing `checkbox` here which would also create a loop :/

Reply:
we can't have a loop here but yes adding the dependency makes sense